query/physicalplan/distinct.go

```diff
@@ -14,7 +14,7 @@ import (
 
 type Distinction struct {
 	pool     memory.Allocator
-	next     func(r arrow.Record) error
+	nextPlan PhysicalPlan
```

Comment:
Changing the callbacks to just be exactly the same as any `PhysicalPlan` is very clever and makes a ton of sense!